Source/Microsoft.Teams.Apps.ExpertFinder/Dialogs/MainDialog.cs

```diff
@@ -79,36 +77,64 @@ public MainDialog(IGraphApiHelper graphApiHelper, IUserProfileActivityStorageHel
                 nameof(OAuthPrompt),
                 new OAuthPromptSettings
                 {
-                    ConnectionName = optionsAccessor.CurrentValue.ConnectionName,
-                    Text = Strings.SigninCardText,
-                    Title = Strings.SignInBtnText,
-                    Timeout = Convert.ToInt32(TimeSpan.FromMinutes(5).TotalMilliseconds), // In milliseconds
+                    ConnectionName = botSettings.CurrentValue.OAuthConnectionName,
+                    Text = Strings.SignInCardText,
+                    Title = Strings.SignInButtonText,
+                    Timeout = (int)TimeSpan.FromMinutes(5).TotalMilliseconds,
                 }));
 
             this.AddDialog(new WaterfallDialog(
-                nameof(WaterfallDialog),
-                new WaterfallStep[] { this.OAuthPromptStepAsync, this.MyProfileAndSearchAsync }));
+                "MainDialog",
+                new WaterfallStep[] { this.CheckForUnknownInputAsync, this.OAuthPromptStepAsync, this.MyProfileAndSearchAsync }));
 
             // The initial child Dialog to run.
-            this.InitialDialogId = nameof(WaterfallDialog);
+            this.InitialDialogId = "MainDialog";
+        }
+
+        /// <summary>
+        /// Check for unknown input, and show the help prompt.
+        /// </summary>
+        /// <param name="stepContext">Provides context for a step in a bot dialog.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
+        /// <returns>Tracking task.</returns>
+        private async Task<DialogTurnResult> CheckForUnknownInputAsync(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            var activity = stepContext.Context.Activity;
+            if (activity.Type == ActivityTypes.Message)
+            {
+                switch (activity.Text?.Trim())
+                {
+                    case Constants.MyProfile:
+                    case Constants.Search:
+                    case null:
```

Comment:
just curious - when would the activity text be null? And if null then it will still check for user sign in.

Reply:
messageBack. I wanted this to be conservative and only catch cases where the user actually typed something wrong.